### DISCUSSION TITLE
UpdateManager warning spam

### DISCUSSION THREAD

**Discussion by TheDenast** [Pryaxis/TShock#2937] [Q&A]
Hi all, I'm a complete noob who has never touched Terraria or hosted a Terraria server. I've recently configured a Tshock 5.1.3.0 server on my Debian bullseye host, and my console log is getting bombarded by the following warning;
```
UpdateManager warning: The request was canceled due to the configured HttpClient.Timeout of 5 seconds elapsing.
        -> A task was canceled.
        -> A task was canceled.
Retrying in 5 minutes.
```
This message completely fills logs logs and console, making looking for any other outputs a big pain. What is the best way to stop this? I'm not a Terraria power user and it does not matter that much to stay up to date, I just want to stop this spam in the easiest way possible, either by fixing the update issue or by simply turning off the warning.

**Comment by sgkoishi** [comment]
The easiest way might be to have a plugin for that - tho I'm not sure if there is a plugin solely for this purpose. ~~I have a plugin containing this, but it also comes with tons of other features that you might not want~~.

If you really don't care about the version, you may try this:
* Click the Fork button on the top right of this page
* Goto `TShockAPI/UpdateManager.cs` and add a `return;` after the `{` and before the `var updates = ...`
https://github.com/Pryaxis/TShock/blob/2c06e2bdf5d3057aaaaf7588519ce9ee305d7079/TShockAPI/UpdateManager.cs#L97-L99
Then you will get a modified version of TShock with update checks disabled. It is a modified version, so if you update to the latest version, update is back, and you need to repeat those steps to disable it again.
